### ISSUE TITLE
Overloads cause incorrect REFL004

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#141]
Overload resolution selects the `IReadOnlyDictionary` overload and therefore there is no problem with this code, but 0.1.11-dev is reporting REFL004 "More than one member is matching the criteria."

```c#
class C
{
    void M(object p) { }

    void M(IReadOnlyDictionary<string, object> p)
    {
        typeof(C).GetMethod(
            nameof(M),
            BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly,
            null,
            new[] { typeof(Dictionary<string, object>) },
            null);
    }
}
```
